airflow/contrib/hooks/snowflake_hook.py

```diff
@@ -69,9 +72,8 @@ def get_uri(self):
         """
         conn_config = self._get_conn_params()
         uri = 'snowflake://{user}:{password}@{account}/{database}/'
-        uri += '{schema}?warehouse={warehouse}'
-        return uri.format(
-            **conn_config)
+        uri += '{schema}?warehouse={warehouse}&role={role}'
```

Comment:
`role` here is an **optional** config.

But users will encounter `KeyError` if they don't explicitly specify `role`, due to this line.

Reply:
@XD-DENG in L65
```python
"role": self.role or role or '',
```
so will not raise `KeyError` I think. but when `role=''` and L75 will be
```python
{schema}?warehouse={warehouse}&role=''
```
also not make sense

Reply:
Thanks @zhongjiajie . Yes, I overlooked L65.
Your point is correct. Hence there is still an issue here.

Reply:
@XD-DENG you are welcome.

Reply:
I tested the uri when role is not supplied. It does not hurt anything to have the empty role=. Though it is not obvious.  Should I add logic to only include that key in URI when val > ''?